src/audio/copier/copier.c

```diff
@@ -1720,18 +1720,23 @@ static uint64_t copier_get_processed_data(struct comp_dev *dev, uint32_t stream_
 {
 	struct copier_data *cd = comp_get_drvdata(dev);
 	uint64_t ret = 0;
-	bool source = dev->direction == SOF_IPC_STREAM_CAPTURE;
 
-	if (cd->endpoint_num && cd->bsource_buffer != input) {
-		if (stream_no < cd->endpoint_num) {
-			if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
-				if (source == input)
-					ret = cd->hd->total_data_processed;
-			} else {
-				ret = comp_get_total_data_processed(cd->endpoint[stream_no],
-								    0, input);
-			}
+	/*
+	 * total data processed is calculated as: accumulate all input data in bytes
+	 * from host, then store in host_data structure, this function intend to retrieve
```

Comment:
ok, but we need to say why we have to chose between the two methods below.  why is it important that playback, type == host is used to select teh calculation.

Reply:
ok， adding more comments.